### DISCUSSION TITLE
Implementing Redis with TLS certificate authentication connection

### DISCUSSION THREAD

**Discussion by jexeq** [adonisjs/core#4330] [Help]
Hi. 

Before anything, I want to thank the Adonis team. It is a great framework, very helpfull and quite stable.

Now with my question:

We work with distributed microservices and dedicated server instances for db and redis connection

In our case, we use TLS certificate authentication to connect our microservices to our redis intance everytime it is required.

for common express microservices, it is not an issue

but in our AdonisJs projects, we couldn't use native adonis redis wrapper library for that porpouse

it seems that last version of the library does not allows TLS connection parameters

I've been looking closer to the library, and, since it wrapes another redis library, that feature was not included in it.

Do anyone knows if exist any way to use TLS certification config for connecting to Redis using native adonis/redis library?

**Comment by sayeed205** [comment]
Any update on it ?

**Reply by jexeq** [reply]
nothing till now. We solved it using directly another library for connecting with redis, skipping built in adonis js config

**Comment by thetutlage** [comment]
Can you please share what you guys have tried so far? An example config?

**Comment by jexeq** [comment]
I can not share the exact code example since it's private of the organization

what you need to do is to register a provider in adonis using npm library:

"redis": "^4.6.7"

follow npm redis library doc and you'll be able to connect using tls certificates

**Comment by thetutlage** [comment]
I needed the AdonisJS redis config to understand why its not working. As per me, the TLS should work outside of the box, since `ioredis` supports and it and we are thin wrapper on top of it.
